<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>magi_llm_window</class>
 <widget class="QMainWindow" name="magi_llm_window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>854</width>
    <height>1033</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>11</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Magi LLM</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="textgenTab">
      <property name="toolTip">
       <string/>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="chat_textgenTab">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>Chat</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="8" column="0">
         <widget class="QPushButton" name="chatGenerateButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <italic>false</italic>
            <bold>true</bold>
            <underline>false</underline>
           </font>
          </property>
          <property name="toolTip">
           <string>Send. Control+Enter shortcut</string>
          </property>
          <property name="text">
           <string>Generate</string>
          </property>
          <property name="shortcut">
           <string>Ctrl+Return</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="1" column="1">
            <widget class="QComboBox" name="awesomePresetComboBox"/>
           </item>
           <item row="0" column="3" colspan="2">
            <widget class="QComboBox" name="characterPresetComboBox">
             <property name="toolTip">
              <string>Select a character</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="instructPresetComboBox">
             <property name="toolTip">
              <string>Select instruct format</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QRadioButton" name="instructRadioButton">
             <property name="text">
              <string>Instruct:</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Awesome prompts:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QRadioButton" name="charactersRadioButton">
             <property name="text">
              <string>Characters:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QTextEdit" name="chat_modeTextHistory">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>Output goes here</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <widget class="QPlainTextEdit" name="chat_modeTextInput">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>128</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="plainText">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Type something here</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="2">
         <widget class="QComboBox" name="chatInputSessionCombo">
          <property name="maxCount">
           <number>128</number>
          </property>
          <property name="placeholderText">
           <string>User input history</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QFrame" name="frame_6">
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="0" column="0">
            <widget class="QPushButton" name="chatContinueButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Continue the last generation</string>
             </property>
             <property name="text">
              <string>Continue</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="chatClearButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Clear the output</string>
             </property>
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
           <item row="0" column="4">
            <widget class="QPushButton" name="chatStopButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Stop generation</string>
             </property>
             <property name="text">
              <string>Stop</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="chatRewindButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Rewind</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="standard_textgenTab">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>Standard</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="1" column="0" colspan="2">
         <widget class="QPlainTextEdit" name="defaultTextInput">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="placeholderText">
           <string>Type something here</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="defaultGenerateButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="toolTip">
           <string>Send. Control+Enter shortcut</string>
          </property>
          <property name="text">
           <string>Generate</string>
          </property>
          <property name="shortcut">
           <string>Ctrl+Return</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QFrame" name="frame_7">
          <layout class="QGridLayout" name="gridLayout_12">
           <item row="0" column="0">
            <widget class="QPushButton" name="defaultContinueButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Continue the last generation</string>
             </property>
             <property name="text">
              <string>Continue</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="defaultClearButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Clear the output</string>
             </property>
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="defaultStopButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Stop generation</string>
             </property>
             <property name="text">
              <string>Stop</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QTextEdit" name="default_modeTextHistory">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>Output appears here</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="notebook_textgenTab">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>Notebook</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <widget class="QPushButton" name="notebookGenerateButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="toolTip">
           <string>Send. Control+Enter shortcut</string>
          </property>
          <property name="text">
           <string>Generate</string>
          </property>
          <property name="shortcut">
           <string>Ctrl+Return</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QFrame" name="frame_4">
          <layout class="QGridLayout" name="gridLayout_9">
           <item row="0" column="1">
            <widget class="QPushButton" name="notebookClearButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Clear the output</string>
             </property>
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QPushButton" name="notebookContinueButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Continue the last generation</string>
             </property>
             <property name="text">
              <string>Continue</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="notebookStopButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Stop</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QTextEdit" name="notebook_modeTextHistory">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="acceptRichText">
           <bool>false</bool>
          </property>
          <property name="placeholderText">
           <string>Type something here</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="settingsTab">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QScrollArea" name="scrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>812</width>
             <height>871</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox_5">
              <property name="title">
               <string>Backend:</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_10">
               <item row="3" column="0">
                <widget class="QPushButton" name="paramWinShowButton">
                 <property name="toolTip">
                  <string>Open parameter window</string>
                 </property>
                 <property name="text">
                  <string>Parameters</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QRadioButton" name="cppCheck">
                 <property name="toolTip">
                  <string>Use llama.cpp backend</string>
                 </property>
                 <property name="text">
                  <string>llama.cpp</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QRadioButton" name="exllamaCheck">
                 <property name="toolTip">
                  <string>Use Exllama backend</string>
                 </property>
                 <property name="text">
                  <string>Exllama</string>
                 </property>
                 <property name="checked">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QRadioButton" name="tsServerCheck">
                 <property name="text">
                  <string>TextSynth</string>
                 </property>
                 <property name="checked">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QCheckBox" name="cppServerCheck">
                 <property name="toolTip">
                  <string>Connects to a launched llama.cpp server instead of llama-cpp-python</string>
                 </property>
                 <property name="text">
                  <string>Server</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QGroupBox" name="groupBox_3">
              <property name="title">
               <string>Themes</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="QRadioButton" name="themeDarkCheck">
                 <property name="text">
                  <string>Dark</string>
                 </property>
                 <property name="checked">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="themeLightCheck">
                 <property name="text">
                  <string>Light</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="themeNativeCheck">
                 <property name="text">
                  <string>Native</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="1" column="0" colspan="2">
             <widget class="QGroupBox" name="groupBox_2">
              <property name="title">
               <string>Settings:</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_5">
               <item row="2" column="0">
                <widget class="QLabel" name="label_28">
                 <property name="text">
                  <string>llama.cpp: Model path:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0" colspan="2">
                <widget class="Line" name="line">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QCheckBox" name="streamEnabledCheck">
                 <property name="toolTip">
                  <string>Stream responses</string>
                 </property>
                 <property name="text">
                  <string>Stream responses</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QToolButton" name="cppModelSelect">
                 <property name="toolTip">
                  <string>Select GGML model</string>
                 </property>
                 <property name="text">
                  <string>...</string>
                 </property>
                 <property name="arrowType">
                  <enum>Qt::NoArrow</enum>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="cppModelPath">
                 <property name="toolTip">
                  <string>Path to GGML model for llama.cpp</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Exllama: Model directory:</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QLineEdit" name="exllamaModelPath">
                 <property name="toolTip">
                  <string>Path to Exllama GPTQ model folder (mode file, tokenizer.json, config.json)</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="3" column="2">
                <widget class="QToolButton" name="exllamaModelSelect">
                 <property name="toolTip">
                  <string>Select Exllama model folder</string>
                 </property>
                 <property name="text">
                  <string>...</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QCheckBox" name="logChatCheck">
                 <property name="toolTip">
                  <string>Write chat logs to file</string>
                 </property>
                 <property name="text">
                  <string>Log chats</string>
                 </property>
                 <property name="checked">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Chat prefixes:</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_7">
               <item row="3" column="0">
                <widget class="QCheckBox" name="customResponsePrefixCheck">
                 <property name="text">
                  <string>Response prefix:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QLineEdit" name="yourNameLine">
                 <property name="toolTip">
                  <string>User name to be used in Character mode</string>
                 </property>
                 <property name="text">
                  <string>User</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="3">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Bot name:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>User name:</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="2">
                <widget class="QLineEdit" name="customResponsePrefix">
                 <property name="toolTip">
                  <string>Prefix to apply to prompts</string>
                 </property>
                 <property name="text">
                  <string>Sure! </string>
                 </property>
                </widget>
               </item>
               <item row="1" column="4">
                <widget class="QLineEdit" name="botNameLine">
                 <property name="text">
                  <string>Assistant</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="settingsPathSaveButton">
          <property name="toolTip">
           <string>Save model paths</string>
          </property>
          <property name="text">
           <string>Save settings</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="llm_menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>854</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSettings"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSettings">
   <property name="text">
    <string>Parameters</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>defaultClearButton</sender>
   <signal>clicked()</signal>
   <receiver>default_modeTextHistory</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>975</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>notebookClearButton</sender>
   <signal>clicked()</signal>
   <receiver>notebook_modeTextHistory</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>975</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
